src/coreclr/jit/optimizer.cpp

```diff
@@ -5466,11 +5466,11 @@ void Compiler::optFindLoops()
         optIdentifyLoopsForAlignment(); // Check if any of the loops need alignment
     }
 
+    optLoopsMarked = true;
+
 #ifdef DEBUG
-    fgDebugCheckLoopTable();
+    optLoopTableValid = true;
```

Comment:
You define and in most places use this not under `#ifdef DEBUG`. Seems like we should be consistent about always putting it under `#ifdef DEBUG` or not.

Reply:
I initially had this `DEBUG` only. Will fix.